### PR TITLE
Remove a string comparison about types

### PR DESCRIPTION
**Pull request by TaKO8Ki**

### CODE REVIEW

File: compiler/rustc_typeck/src/check/fn_ctxt/suggestions.rs [hunk 1]
```diff
@@ -238,25 +237,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
-                                );
-                            }
-                        }
-                    }
+                let ty = found_substs.type_at(0);
+                let peeled = ty.peel_refs();
+                if let ty::Adt(adt, _) = peeled.kind()
+                    && self.tcx.is_diagnostic_item(sym::String, adt.did())
+                {
+                    let ref_cnt = ty.to_string().len() - peeled.to_string().len();
```
Comment (compiler-errors) [inline compiler/rustc_typeck/src/check/fn_ctxt/suggestions.rs:252]: Do you mind removing this string comparison too? I think it's checking how many `ty::Ref` we are wrapped in. You can probably count this by replacing `ty.peel_refs()` with a `while let`, something like:

```
let mut peeled = ty;
while let ty::Ref(_, inner, _) = peeled.kind() {
  peeled = *inner;
  ref_cnt += 1;
}
```

Comment (TaKO8Ki) [reply compiler/rustc_typeck/src/check/fn_ctxt/suggestions.rs]: I fixed that!

**Comment by compiler-errors** [comment]
Thanks. This looks good now.

@bors r+

**Pull request merged**